curriculum/challenges/english/15-javascript-algorithms-and-data-structures-22/learn-intermediate-fetch-promises-and-async-await-by-building-an-fcc-forum-leaderboard/643c90830bbeb40ab15148a6.md

```diff
@@ -0,0 +1,169 @@
+---
+id: 643c90830bbeb40ab15148a6
+title: Step 30
+challengeType: 0
+dashedName: step-30
+---
+
+# --description--
+
+`forEach` is an array method in JavaScript that allows you to iterate over each element of an array and perform some operation on each element. Here's an example:
+
+```js
+const arr = [1, 2, 3, 4, 5];
+arr.forEach((element) => {
+  console.log(element);
+});
+```
+
+In this step, you will use the `forEach` method on the `topics` array, Like this 
+
+```js
+topics.forEach();
+```
+
+# --hints--
+
+You should use the `forEach` method on the `topics` array.
+
+```js
+assert(code.match(/topic_list;\s*topics\.forEach\(/g));
```

Comment:
```suggestion
assert.match(code, /topic_list;\s*topics\.forEach\(/);
```

We try to use the most accurate method on the assert API (in this case `.match`), rather than a raw `assert()`.

Also, test regex shouldn't be global unless we're specifically testing the number of matches, not just that a match exists.

Reply:
fixed